plugins/osdn/podinfoserver.go

```diff
@@ -0,0 +1,138 @@
+package osdn
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/gorilla/mux"
+
+	"github.com/openshift/openshift-sdn/pkg/netutils"
+	"github.com/openshift/openshift-sdn/plugins/osdn/api"
+)
+
+type PodInfoServer struct {
+	http.Server
+
+	registry *Registry
+	vnidMap  *VNIDMap
+	nodeName string
+
+	clusterNetwork string
+	nodeNetwork    string
+	nodeGateway    string
+	mtu            uint
+}
+
+func NewPodInfoServer(r *Registry, v *VNIDMap, nodeName string) *PodInfoServer {
+	router := mux.NewRouter()
+ 
+	s := &PodInfoServer{
+		registry: r,
+		vnidMap:  v,
+		nodeName: nodeName,
+		Server:   http.Server{
+			Handler: router,
+		},
+	}
+
+	router.NotFoundHandler = http.HandlerFunc(http.NotFound)
+	router.Methods("GET").Path("/network").HandlerFunc(s.getNetwork)
```

Comment:
I am *really* uncomfortable about creating a new API here (which is what this is) and also not happy to see a new set of API definitions that don't overlap with our existing objects being created.  I need to dive into this more and understand the tradeoffs, but like to see a quick writeup of what options you considered and why you chose this approach.

One question I was surprised by - why does the CNI interface not pass enough information down to the network plugin to make a decision there?